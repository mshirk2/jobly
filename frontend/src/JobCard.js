import React from "react";
import { Link } from "react-router-dom";

function JobCard({id, title, salary, equity, companyHandle}) {
    return (
        <Link className="JobCard" to={`/jobs/${id}`}>
            <div>
                <h5>{title}</h5>
                <p>{companyHandle}</p>
                <p>Salary: ${salary}</p>
                <p>Equity: {equity}</p>
            </div>
        </Link>
    );
}

export default JobCard;