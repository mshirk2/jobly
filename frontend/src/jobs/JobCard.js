import React from "react";
import { Link } from "react-router-dom";
import formatSalary from "../hooks/useFormatSalary";
import placeholderLogo from "../images/placeholderLogo.png"
import "./JobCard.css"

function JobCard({id, title, salary, equity, companyName, companyHandle, logoUrl}) {
    if(!logoUrl) logoUrl = placeholderLogo;

    return (
        <div className="JobCard card">
            <div className="card-body">
                <Link to={`/jobs/${id}`} className="link">
                    {logoUrl && <img src={logoUrl} alt={companyName} className="logo"/>}
                    <h5 className="card-title">{title}</h5>
                </Link>
                <Link to={`/companies/${companyHandle}`} className="link">
                    <p className="card-subtitle">{companyName}</p>
                </Link>
                {salary && <div>Salary: {formatSalary(salary)}</div>}
                {equity > 0 && <div>Equity: {equity}</div>}
            </div>
        </div>
    );
}

export default JobCard;