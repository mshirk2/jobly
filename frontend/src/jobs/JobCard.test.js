import React from "react";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";
import { MemoryRouter } from "react-router";

it('renders without crashing', () =>{
    render(
        <MemoryRouter>
            <JobCard />
        </MemoryRouter>
    )
});

it ("matches snapshot", () =>{
    const { asFragment } = render(
        <MemoryRouter>
            <JobCard 
                title="App Tester"
                salary="50000"
                equity="0.05"
            />
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
});