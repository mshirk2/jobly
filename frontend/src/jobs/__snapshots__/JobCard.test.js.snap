// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCard card"
  >
    <div
      class="card-body"
    >
      <a
        class="link"
        href="/jobs/undefined"
      >
        <img
          class="logo"
          src="placeholderLogo.png"
        />
        <h5
          class="card-title"
        >
          App Tester
        </h5>
      </a>
      <a
        class="link"
        href="/companies/undefined"
      >
        <p
          class="card-subtitle"
        />
      </a>
      <div>
        Salary: $50,000
      </div>
      <div>
        Equity: 0.05
      </div>
    </div>
  </div>
</DocumentFragment>
`;
